File: app/src/main/java/com/snaptiongame/app/data/providers/UserProvider.java
package com.snaptiongame.app.data.providers;

import com.snaptiongame.app.data.api.SnaptionApi;
import com.snaptiongame.app.data.auth.AuthManager;
import com.snaptiongame.app.data.models.Friend;
import com.snaptiongame.app.data.models.User;
import com.snaptiongame.app.data.models.UserStats;
import com.snaptiongame.app.data.providers.api.ApiProvider;

import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Single;

/**
 * @author Tyler Wong
 */

public class UserProvider {
    private static SnaptionApi apiService = ApiProvider.getApiService();

    public static Single<User> getUser(int userId) {
        return apiService.getUser(userId);
    }

    public static Observable<List<User>> searchUsers(String email, String facebookId,
                                                     String username, String fullName, int page) {
        return apiService.searchUsers(email, facebookId, username, fullName, page);
    }

    public static Single<User> updateUser(User user) {
        return apiService.updateUser(user);
    }
    
    public static Single<UserStats> getUserStats(int userId) {
        return apiService.getUserStats(userId);
    }
}

File: app/src/main/java/com/snaptiongame/app/data/providers/UserProvider.kt
@file:JvmName("UserProvider")

package com.snaptiongame.app.data.providers

import com.snaptiongame.app.data.models.User
import com.snaptiongame.app.data.models.UserStats
import com.snaptiongame.app.data.providers.api.ApiProvider
import io.reactivex.Flowable

import io.reactivex.Single

/**
 * @author Tyler Wong
 */

private val apiService = ApiProvider.getApiService()

fun getUser(userId: Int): Single<User> = apiService.getUser(userId)

fun searchUsers(email: String, facebookId: String,
                username: String, fullName: String, page: Int): Flowable<List<User>> =
    apiService.searchUsers(email, facebookId, username, fullName, page)

fun updateUser(user: User): Single<User> = apiService.updateUser(user)

fun getUserStats(userId: Int): Single<UserStats> = apiService.getUserStats(userId)
